src/chat.rs

```diff
@@ -1722,34 +1722,34 @@ pub unsafe fn set_chat_profile_image(
 ) -> Result<(), Error> {
     ensure!(chat_id > DC_CHAT_ID_LAST_SPECIAL, "Invalid chat ID");
 
-    let mut OK_TO_CONTINUE = true;
     let mut success = false;
 
     let mut chat = Chat::load_from_db(context, chat_id)?;
     let mut msg = dc_msg_new_untyped(context);
     let mut new_image_rel = None;
 
-    if real_group_exists(context, chat_id) {
-        if !(is_contact_in_chat(context, chat_id, 1i32 as u32) == 1i32) {
-            log_event!(
-                context,
-                Event::ERROR_SELF_NOT_IN_GROUP,
-                0,
-                "Cannot set chat profile image; self not in group.",
-            );
-        } else {
-            /* we should respect this - whatever we send to the group, it gets discarded anyway! */
+    (|| {
```

Comment:
uff, can we not use this kind of construct please, it makes it much harder to read

Reply:
what do you suggest?

Reply:
Now better?